pkg/doc/signature/ed25519signature2018/public_key_verifier.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ed25519signature2018
+
+import (
+	"crypto/ed25519"
+	"errors"
+	"fmt"
+
+	sigverifier "github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+)
+
+// PublicKeyVerifier verifies a Ed25519 signature taking Ed25519 public key bytes as input.
+type PublicKeyVerifier struct {
+}
+
+// Verify will verify a signature.
+func (v *PublicKeyVerifier) Verify(pubKeyValue interface{}, doc, signature []byte) error {
+	pubKey, ok := pubKeyValue.(*sigverifier.PublicKey)
```

Comment:
This function should be calling a KMS implementation, not converting interfaces here... Is that the plan?

Reply:
`NewCryptoVerifier()` is what we would like users to use. `PublicKeyVerifier` is a sort-of backward compatibility struct which is used in other packages (`verifiable`, `did`, `didexchange`,  `legacykms`) and should be replaced in the follow-up PRs. I can leave a comment regarding this in the code.